### PR TITLE
[NETBEANS-5523] PHP 8.0: Treat namespaced names as single token

### PR DESCRIPTION
**Pull request by junichi11**
We can fix the highlighting for keyword parts of namespaced names if we add new tokens.
However, many existing features using the lexer will be broken.
So, We won't fix it at the moment.

- https://issues.apache.org/jira/browse/NETBEANS-5523
- https://wiki.php.net/rfc/namespaced_names_as_token

- Fix the parser and the lexer
- Fix/Add unit tests

nb-php80-treat-namespaced-names-as-single-token

### CODE REVIEW

**Comment by junichi11** [comment]
@tmysik Could you please have a look at this when you have time? If there is no problem, let's merge it. Thanks!

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/NamespaceName.java [hunk 1]
```diff
@@ -64,6 +70,24 @@ public NamespaceName(int start, int end, List segments, boolean global, boolean
         this.current = current;
     }
 
+    @NonNull
+    public static NamespaceName create(int start, int end, @NonNull String namespaceName) {
+        boolean isGlobal = namespaceName.startsWith("\\"); // NOI18N
+        boolean isCurrent = namespaceName.startsWith("namespace\\"); // NOI18N
+        String[] names = namespaceName.split("\\\\"); // NOI18N
+        int startSegment = start;
+        List<Identifier> list = new ArrayList<>();
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/NamespaceName.java:79]: Nitpick:
```suggestion
        List<Identifier> list = new ArrayList<>(names.size());
```

Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/NamespaceName.java:79]: Will fix it next time :) Thanks!

**Comment by junichi11** [comment]
@tmysik Thank you for your review!

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/NamespaceName.java (hunk 1 above)
Comment (tmysik) [reply php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/NamespaceName.java:79]: Sure, it is really a nitpick :)

**Pull request merged**